### PR TITLE
Jlopezbarb/tty support

### PR DESCRIPTION
**Pull request by jLopezbarb**
Adds support for tty on deploy commands

## Proposed changes
- Adds tty support
-

### CODE REVIEW

File: cmd/deploy/deploy.go [hunk 1]
```diff
@@ -205,13 +208,33 @@ func (dc *deployCommand) runDeploy(ctx context.Context, cwd string, opts *Option
...
+				return
+			}
+		}
+		exit <- nil
+	}()
+
+	select {
+	case <-stop:
+		log.Infof("CTRL+C received, starting shutdown sequence")
+		dc.cleanUp(ctx)
```
Comment (ifbyol) [inline cmd/deploy/deploy.go:229]: Wouldn't the defer command at line 196 would be executed before the return?

File: cmd/deploy/deploy.go [hunk 2]
```diff
@@ -205,13 +208,33 @@ func (dc *deployCommand) runDeploy(ctx context.Context, cwd string, opts *Option
...
+		return errors.ErrIntSig
+	case err := <-exit:
+		dc.cleanUp(ctx)
```
Comment (ifbyol) [inline cmd/deploy/deploy.go:232]: Same about the defer

File: cmd/deploy/deploy.go [hunk 3]
```diff
@@ -205,13 +208,33 @@ func (dc *deployCommand) runDeploy(ctx context.Context, cwd string, opts *Option
...
+			if err := dc.executor.Execute(command, opts.Variables); err != nil {
+				log.Infof("error executing command '%s': %s", command, err.Error())
+				exit <- fmt.Errorf("error executing command '%s': %s", command, err.Error())
+				return
+			}
+		}
+		exit <- nil
+	}()
+
+	select {
```
Comment (ifbyol) [inline cmd/deploy/deploy.go:226]: Wouldn't this be needed also for the execution of destroy commands?

**Review by ifbyol — COMMENTED** [review]
Can you add a video on how the tty output would look like?

**Comment by pchico83** [comment]
@jLopezbarb if `tty` is selected, `okteto build` commands inside a pipeline should be shown in `tty`

**Comment by jLopezbarb** [comment]
> @jLopezbarb if `tty` is selected, `okteto build` commands inside a pipeline should be shown in `tty`

If we show them in tty we can't collapse when finalising the command because both commands (okteto build and okteto deploy) writes on stdout at the same time

**Comment by jLopezbarb** [comment]
The code is quite outdated and the conflicts generated are more complex than redoing it, I'm going to open this PR again based on main

**Pull request closed without merging**